aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -563,28 +564,57 @@ SparseTensor& add_out_sparse_cpu(const SparseTensor& t, const SparseTensor& src,
   TORCH_CHECK(!r.is_cuda(), "add: expected 'out' to be CPU tensor, but got CUDA tensor");
   TORCH_CHECK(!src.is_cuda(), "add: expected 'other' to be a CPU tensor, but got a CUDA tensor");
 
-  TORCH_CHECK(t.sizes().equals(src.sizes()), "add: expected sizes of 'self' and 'other' to match, but ", t.sizes(), " != ", src.sizes());
-
   auto commonDtype = promoteTypes(t.scalar_type(), src.scalar_type());
 
   TORCH_CHECK(canCast(commonDtype, r.scalar_type()), "Can't convert result type ", commonDtype, " to output ", r.scalar_type(), " in add operation");
 
+  // Check if the broadcasting can be performed and if applicable return the shape of the result.
+  const std::vector<int64_t>& res_shape = infer_size(t.sizes(), src.sizes());
+  bool is_same_size = t.sizes().equals(src.sizes());
+
+  // deal with empty sparse tensors
   if (src._nnz() == 0) {
-    return copy_sparse_to_sparse_(r, t);
+    if (is_same_size) {
+      return copy_sparse_to_sparse_(r, t);
+    } else {
+    const SparseTensor& broadcasted_t = sparse_broadcast_to(t, res_shape);
+    return copy_sparse_to_sparse_(r, broadcasted_t);
```

Comment:
Nit: use proper indentation here and below.

Also, I suggest using the same approach of handling `is_same_size` as you have used in SparseCUDATensorMath.cu:
```
if (is_same_size) {
  broadcasted_t = t;
  ...
} else {
  broadcasted_t = sparse_broadcast_to(t, res_shape);
  ...
}
```
and in the following replace `t` with `broadcasted_t` (or call it `adjusted_t` or similar).

Reply:
Nesting these conditions is a bit convoluted. Why not just always broadcast `t` to `res_shape` and copy only on a single condition of `src._nnz() == 0`?